reana_workflow_engine_serial/tasks.py

```diff
@@ -136,14 +136,16 @@ def run(workflow_json,
                            publisher)
 
     for step_number, step in enumerate(steps_to_run):
-        run_step(step_number,
-                 step,
-                 workflow_workspace,
-                 cache_enabled,
-                 expanded_workflow_json,
-                 workflow_json,
-                 publisher,
-                 workflow_uuid)
+        status = run_step(step_number,
+                          step,
+                          workflow_workspace,
+                          cache_enabled,
+                          expanded_workflow_json,
+                          workflow_json,
+                          publisher,
+                          workflow_uuid)
+        if job_status.status != 'succeeded':
+            break
```

Comment:
Is the job status properly propagated?  I'm getting `finished` for a workflow where the first step contains a shells command typo error (`mdir` instead of `mkdir`):

```console
$ reana-client status -w fail
NAME   RUN_NUMBER   CREATED               STATUS     PROGRESS
fail   1            2020-01-13T12:25:14   finished   1/3
```

I would have expected to  get `failed` instead of `finished`.

Reply:
... whoops! Must have been on a different branch when I tried.... It works well :+1: